core/src/main/scala/cats/Invariant.scala

```diff
@@ -105,6 +105,10 @@ object Invariant extends ScalaVersionSpecificInvariantInstances with InvariantIn
   implicit def catsInstancesForId
     : Distributive[Id] with Bimonad[Id] with CommutativeMonad[Id] with NonEmptyTraverse[Id] =
     cats.catsInstancesForId
+  @deprecated("Added for bincompat", "2.8.0")
+  @cats.compat.targetName("catsInstancesForId")
+  def catsInstancesForIdCompat2_6_1: Comonad[Id] =
+    cats.catsInstancesForId
```

Comment:
This is the fix for https://github.com/typelevel/cats/issues/4062. Added `@targetName` as a dummy annotation in Scala 2 so we don't have to split this file (I tried compat `trait`s first and it didn't work).